src/memory.rs

```diff
@@ -0,0 +1,53 @@
+use crate::constants::MEMORY_MAX;
+
+pub struct Memory {
+    memory: [u16; MEMORY_MAX as usize],
+}
+
+impl Memory {
+    pub fn read(&self, addr: u16) -> u16 {
+        self.memory.get(addr as usize).unwrap().clone()
+    }
+
+    pub fn write(&mut self, addr: u16, data: u16) {
+        *self.memory.get_mut(addr as usize).unwrap() = data;
+    }
+}
+
+impl Default for Memory {
+    fn default() -> Self {
+        Self {
+            memory: [0; MEMORY_MAX as usize],
+        }
+    }
+}
```

Comment:
This is equivalent to deriving `Default`, which is less boilerplate.

Reply:
I originally thought so! But when I do, I get the following error:

```sh
error[E0277]: the trait bound `[u16; 65535]: Default` is not satisfied
 --> src/memory.rs:5:5
  |
3 | #[derive(Default)]
  |          ------- in this derive macro expansion
4 | pub struct Memory {
5 |     memory: [u16; MEMORY_MAX as usize],
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Default` is not implemented for `[u16; 65535]`
  |
  = help: the following other types implement trait `Default`:
            &[T]
            &mut [T]
            [T; 0]
            [T; 10]
            [T; 11]
            [T; 12]
            [T; 13]
            [T; 14]
          and 27 others
  = note: this error originates in the derive macro `Default` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
Maybe you can use a `Vec<u16>` for `memory`?